### PR TITLE
BGE Flow data-qa-locator attributes

### PR DESCRIPTION
**Pull request by bdvorachek**
* Added `data-qa-locator` attributes to items in the batch gift entry flow

# Critical Changes

# Changes
- Updated Add New Allocation button to utilize a custom label
# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/lwc/geBatchWizard/geBatchWizard.html [hunk 1]
```diff
@@ -67,10 +68,9 @@ <h2 class="slds-modal__title">
                                                        size="12">
                                     <c-util-expandable-section id={section.id}
                                                                label={section.label}>
-
                                         <template iterator:it={section.elements}>
 
-                                            <div key={it.value.apiName}
+                                            <div key={it.value.id}
```
Comment (bdvorachek) [inline src/lwc/geBatchWizard/geBatchWizard.html:73]: This is intentional and fixes a console warning that appears. apiName is undefined always

File: src/lwc/geBatchWizard/geBatchWizard.html [hunk 2]
```diff
@@ -81,8 +81,7 @@ <h2 class="slds-modal__title">
                                                     <div class="slds-small-size_1-of-1 slds-medium-size_1-of-3 slds-large-size_1-of-3"></div>
                                                 </template>
                                                 <div class="slds-small-size_1-of-1 slds-medium-size_2-of-3 slds-large-size_2-of-3">
-                                                    <c-util-input key={it.value.apiName}
```
Comment (bdvorachek) [inline src/lwc/geBatchWizard/geBatchWizard.html:84]: Similar here, no need for the `key` attribute, the div above takes care of it.

File: src/lwc/geBatchWizard/geBatchWizard.js [hunk 3]
```diff
@@ -258,7 +258,7 @@ export default class geBatchWizard extends NavigationMixin(LightningElement) {
 
         this.selectedBatchHeaderFields = addKeyToCollectionItems(this.selectedTemplate.batchHeaderFields);
         this.formSections = this.selectedTemplate.layout.sections;
-
+        console.log(JSON.parse(JSON.stringify(this.formSections)));
```
Comment (psadie-sfdo) [inline src/lwc/geBatchWizard/geBatchWizard.js:261]: Just a console log to clear

File: src/lwc/geFormField/geFormField.js [hunk 4]
```diff
@@ -127,6 +127,11 @@ export default class GeFormField extends LightningElement {
             this.value = defaultValue;
             this.handlePicklistChange();
         }
+        if(this.targetFieldName) {
```
Comment (psadie-sfdo) [inline src/lwc/geFormField/geFormField.js:130]:
```suggestion
        if (this.targetFieldName) {
```

File: src/lwc/utilInput/utilInput.js [hunk 5]
```diff
@@ -43,6 +43,13 @@ export default class utilInput extends LightningElement {
 
     @track isRichTextValid = true;
 
+    connectedCallback() {
+        // if no data-qa-locator attribute set, populate and reflect to template
+        if(isEmpty(this.getAttribute('data-qa-locator'))) {
```
Comment (psadie-sfdo) [inline src/lwc/utilInput/utilInput.js:48]:
```suggestion
        if (isEmpty(this.getAttribute('data-qa-locator'))) {
```

File: src/lwc/geBatchWizard/geBatchWizard.js [hunk 6]
```diff
@@ -395,4 +395,24 @@ export default class geBatchWizard extends NavigationMixin(LightningElement) {
         this.hasInvalidBatchFields = false;
         this.missingBatchHeaderFieldLabels = [];
     }
+
```
Comment (psadie-sfdo) [inline src/lwc/geBatchWizard/geBatchWizard.js:398]: This comment is more of an open question/conversation. What're your thoughts around adding jsdocs when these qa locator getters start and end? I personally like them if only to immediately call out the group when we're strolling through these components in the future.

**Pull request merged**